lib/job.js

```diff
@@ -104,10 +123,95 @@ function _addTasksWithLen(info, db_entry, storage, len, split_size) {
 }
 
 /**
+ * Populate the response with the task
+ * according to the instance and step and job.
+ */
+function getTask(client,job, storage ,res)
+{
+    //TODO update the prev jobs of client
+    client.prev_jobs.set(job);
```

Comment:
I think this would set `prev\_jobs` to be the complete job object but further down you assume it's a list and .pop() from it. also we probably need to just save something like `job\_id:step` as the elements of list.